### PR TITLE
Add Client type

### PR DESCRIPTION
**Pull request by Kegsay**
Manually tested with my open openid token and it works on matrix.org.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	err = fmt.Errorf("no address found for matrix host %v", host)
+	for _, addr := range dnsResult.Addrs {
+		u := makeHTTPSURL(r.URL, addr)
+		r.URL = &u
+		resp, err = f.transport.RoundTrip(r)
+		if err == nil {
+			return resp, nil
+		}
+	}
+	return nil, err
```
Comment (NegativeMjark) [inline client.go]: shouldn't the ``err = fmt.Errorf`` be here rather than before the for loop?

File: client.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+			// as the empty string so that crypto/tls doesn't add SNI.
+			DialTLS: func(network, addr string) (net.Conn, error) {
+				rawconn, err := net.Dial(network, addr)
+				if err != nil {
+					return nil, err
+				}
+				// Wrap a raw connection ourselves since tls.Dial defaults the SNI
+				conn := tls.Client(rawconn, &tls.Config{
+					ServerName:         "",
+					InsecureSkipVerify: true,
```
Comment (NegativeMjark) [inline client.go]: We should be checking that the TLS certificate we see here matches one of the allowed SHA-256 fingerprints for the server.

That might be a bit much for the initial implementation, but a ``TODO`` comment might be nice.

Comment (Kegsay) [inline client.go]: Done.

File: client.go (hunk 1 above)
Comment (Kegsay) [inline client.go]: Done.

**Pull request merged**